lib/internal/streams/destroy.js

```diff
@@ -36,13 +36,20 @@ function destroy(err, cb) {
 
   this._destroy(err || null, (err) => {
     if (!cb && err) {
-      if (!this._writableState) {
-        process.nextTick(emitErrorAndCloseNT, this, err);
-      } else if (!this._writableState.errorEmitted) {
-        this._writableState.errorEmitted = true;
-        process.nextTick(emitErrorAndCloseNT, this, err);
-      } else {
+      const errorEmitted =
+        (this._writableState && this._writableState.errorEmitted) ||
+        (this._readableState && this._readableState.errorEmitted);
+
+      if (errorEmitted) {
         process.nextTick(emitCloseNT, this);
+      } else {
+        if (this._writableState) {
+          this._writableState.errorEmitted = true;
+        }
+        if (this._readableState) {
+          this._readableState.errorEmitted = true;
+        }
+        process.nextTick(emitErrorAndCloseNT, this, err);
       }
```

Comment:
this is so `destroy()` work with `Readable` as well in terms of emitting `error`.